docs/rfcs/sync-manager.md

```diff
@@ -0,0 +1,171 @@
+
+* The sync manager is responsible for managing the "clients" collection - engines should never see details about that engine. However, engines will need the ability to be told about commands targetting that engine. The sync manager will also need to be able to accept commands to be delivered to other clients - although that needs more thought as the conditions in which a command need to be sent typically aren't managed by the engine itself - eg, when a "bookmark" restore is done.
+
+* Each engine is responsible for persisting engine-specific stuff which needs to be persisted. An obvious thing which needs to be persisted is the syncID and version string for the engine itself, so it can take the necessary action when these change.
+
+* The sync manager will manage keys and pass the relevant key into the engine. Keys for engines should not be exposed to engines other than the key it is for (although currently, in practice, all engines share the same key, but this is an implementation detail.
+
+* The sync manager will pass the relevant part of info/collections to the engine and a token which can be used to access storage. Specific engines never talk to the token server.
+
+# Details
+## Structure definitions
+
+While we use rust struct definitions here, it's important to keep in mind that as mentioned above, we'll need to support the manager being written in something other than rust. As the data is relatively small, it may (or may not) be desirable to use JSON as the serialization format.
+
+    // The main structure passed to a sync implementation. This needs to carry
+    // all state necessary to sync.
+    struct GlobalEngineState {
+        // note: no "declined" or "enabled" here - that's invisible to the engine.
+
+        // the result of info/configuration.
+        config: GlobalConfig,
+
+        // The token server token.
+        storage_token: Vec[u8] (??), // not clear on the type here, but whatever.
+
+        // Info about the collections managed by this engine. In most cases there
+        // will be exactly 1.
+        collections: HashMap<String, CollectionState>
+    }
+
+    // The state for a collection. Most engines will manage a single collection, but
+    // there may be many.
+    struct CollectionState {
+        // The info/collections response for this collection.
+        collection_info: CollectionInfo,
+
+        // The current syncID for the engine, as read from meta/global. May be None.
+        // Engines are responsible for persisting this and taking action when it differs.
+        sync_id: Option<SyncGuid>,
+
+        // The keys to be used for this collection. This may be the default key (ie,
+        // used for all collections) or one specific to the collection, but the
+        // engine doesn't need to know that.
+        keys: KeyBundle,
+
+        // Commands the engine should act upon before syncing.
+        commands: Vec<Command>
+    }
+
+     struct Command {
```

Comment:
µnit: enum realistically, maybe.

Reply:
Added a comment here noting it may be an enum